compiler/noirc_evaluator/src/ssa/opt/remove_bit_shifts.rs

```diff
@@ -197,23 +197,20 @@ impl Context<'_, '_, '_> {
                 let zero =
                     self.numeric_constant(FieldElement::zero(), NumericType::signed(bit_size));
                 let lhs_sign = self.insert_binary(lhs, BinaryOp::Lt, zero);
-                let lhs_sign_as_field = self.insert_cast(lhs_sign, NumericType::NativeField);
-                let lhs_as_field = self.insert_cast(lhs, NumericType::NativeField);
-                // For negative numbers, convert to 1-complement using wrapping addition of a + 1
-                // Unchecked add as these are fields
-                let one_complement = self.insert_binary(
-                    lhs_sign_as_field,
-                    BinaryOp::Add { unchecked: true },
-                    lhs_as_field,
-                );
-                let one_complement = self.insert_truncate(one_complement, bit_size, bit_size + 1);
```

Comment:
I'm not a fan of us allowing values to escape the bounds implied by their type as we could have logic elsewhere in the compiler making assumptions based on type information which this would then break.

Casting to `Field` in this case ensures that this does not happen and is safer.

Reply:
In `expand_signed_checks` we also perform some math in signed types, then truncate to the max bit size (which in my mind should be a no-op). I tried casting to Field there too before doing the math but it doesn't seem to work or to be a straight-forward change, so for now I'll drop this PR.